cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetector.java

```diff
@@ -79,7 +82,17 @@ public AnomalyDetector(KafkaCruiseControl kafkaCruiseControl,
                                                        .thenComparingLong(Anomaly::detectionTimeMs));
     KafkaCruiseControlConfig config = kafkaCruiseControl.config();
     _adminClient = KafkaCruiseControlUtils.createAdminClient(KafkaCruiseControlUtils.parseAdminClientConfigs(config));
-    _anomalyDetectionIntervalMs = config.getLong(KafkaCruiseControlConfig.ANOMALY_DETECTION_INTERVAL_MS_CONFIG);
+    long anomalyDetectionIntervalMs = config.getLong(KafkaCruiseControlConfig.ANOMALY_DETECTION_INTERVAL_MS_CONFIG);
+    _goalViolationDetectionIntervalMs = config.getLong(KafkaCruiseControlConfig.GOAL_VIOLATION_DETECTION_INTERVAL_MS_CONFIG) != null ?
+                                        config.getLong(KafkaCruiseControlConfig.GOAL_VIOLATION_DETECTION_INTERVAL_MS_CONFIG) :
+                                        anomalyDetectionIntervalMs;
+    _metricAnomalyDetectionIntervalMs = config.getLong(KafkaCruiseControlConfig.METRIC_ANOMALY_DETECTION_INTERVAL_MS_CONFIG) != null ?
+                                        config.getLong(KafkaCruiseControlConfig.METRIC_ANOMALY_DETECTION_INTERVAL_MS_CONFIG) :
+                                        anomalyDetectionIntervalMs;
+    _diskFailureDetectionIntervalMs = config.getLong(KafkaCruiseControlConfig.DISK_FAILURE_DETECTION_INTERVAL_MS_CONFIG) != null ?
+                                      config.getLong(KafkaCruiseControlConfig.DISK_FAILURE_DETECTION_INTERVAL_MS_CONFIG) :
```

Comment:
Nit: Can we avoid calling `getLong` twice in case the value is not `null` -- please see `SelfHealingNotifier#configure`?